lib/pure/os.nim

```diff
@@ -1634,6 +1634,22 @@ proc setFilePermissions*(filename: string, permissions: set[FilePermission]) {.
       var res2 = setFileAttributesA(filename, res)
     if res2 == - 1'i32: raiseOSError(osLastError(), $(filename, permissions))
 
+const hasCCopyfile = defined(osx) # since osx 10.5
```

Comment:
Maybe add a `or defined(nimLegacyCopyFile)` for those that want to target OSX that's older than 10.5 :)

Reply:
done; future work can allow `when osxVersion >= (10, 5)` by querying at CT the host version of OSX (which could also be set eg for cross compiling) so that `-d:nimLegacyCopyFile` wouldn't be needed